src/main/java/sg/dex/starfish/impl/squid/DexResolver.java

```diff
@@ -0,0 +1,128 @@
+package sg.dex.starfish.impl.squid;
+
+import com.oceanprotocol.common.helpers.EncodingHelper;
+import com.oceanprotocol.squid.api.config.OceanConfig;
+import com.oceanprotocol.squid.api.config.OceanConfigFactory;
+import com.oceanprotocol.squid.exceptions.DIDFormatException;
+
+import org.web3j.protocol.core.methods.request.EthFilter;
+import org.web3j.abi.EventEncoder;
+import org.web3j.protocol.core.DefaultBlockParameter;
+import org.web3j.crypto.CipherException;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import sg.dex.starfish.Resolver;
+import sg.dex.starfish.util.DID;
+import sg.dex.starfish.util.Hex;
+import sg.dex.starfish.util.Utils;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import io.reactivex.Flowable;
+import java.util.Properties;
+
+import com.oceanprotocol.common.web3.KeeperService;
+import com.oceanprotocol.keeper.contracts.DIDRegistry;
+
+public class DexResolver implements Resolver {
+    private DIDRegistry contract;
+    private SquidService squidService;
+
+    /**
+     * Create DexResolver
+     *
+     * @param DIDRegistry contract
+     * @param SquidService squidService
+     */
+    private DexResolver(DIDRegistry contract, SquidService squidService)  {
+        this.contract = contract;
+        this.squidService = squidService;
+    }
+
+    /**
+     * Creates a SquidResolverImpl
+     *
+     * @param String configFile. All information about account credentials will be taken from this file
+     * @throws IOException, CipherException
+     * @return SquidResolverImpl The newly created SquidResolverImpl
+     */
+    public static DexResolver create(String configFile) throws IOException, CipherException{
+        SquidService squidService = SquidService.create(configFile);
+        Properties properties = squidService.getProperties();
+        String address = (String)properties.getOrDefault("contract.DIDRegistry.address", "");
+        OceanConfig oceanConfig = OceanConfigFactory.getOceanConfig(properties);
+        KeeperService keeper = squidService.getKeeperService(oceanConfig);
+        DIDRegistry contract = DIDRegistry.load(address, keeper.getWeb3(), keeper.getTxManager(), keeper.getContractGasProvider());
+        return new DexResolver(contract, squidService);
+    }
+
```

Comment:
We should have integration tests that register assets on-chain, not just the DID/DDO. This may need some DEP specification to define how asset IDs etc get embedded in the DDO.